src/engine/controls/cuecontrol.cpp

```diff
@@ -1800,30 +2087,128 @@ void CueControl::hotcueFocusColorNext(double v) {
     pCue->setColor(colorPalette.nextColor(*color));
 }
 
+void CueControl::setCurrentSavedLoopControl(HotcueControl* pControl) {
+    if (m_pCurrentSavedLoopControl && m_pCurrentSavedLoopControl != pControl) {
+        // Disable previous saved loop
+        m_pCurrentSavedLoopControl->setStatus(HotcueControl::Status::Valid);
+        m_pCurrentSavedLoopControl = nullptr;
+    }
+
+    // Set new control as active
+    if (!pControl) {
+        return;
+    }
+
+    QMutexLocker lock(&m_mutex);
+    if (!m_pLoadedTrack) {
+        return;
+    }
+
+    CuePointer pCue(pControl->getCue());
+
+    // Need to unlock before emitting any signals to prevent deadlock.
+    lock.unlock();
+
+    VERIFY_OR_DEBUG_ASSERT(pCue &&
+            pCue->getType() == mixxx::CueType::Loop &&
+            pCue->getEndPosition() != Cue::kNoPosition) {
+        return;
+    }
+
+    m_pCurrentSavedLoopControl = pControl;
+    qDebug() << "CueControl::setLoop" << pCue->getPosition()
+             << pCue->getEndPosition();
+    setLoop(pCue->getPosition(), pCue->getEndPosition(), true);
+    pControl->setStatus(HotcueControl::Status::Active);
+}
+
+void CueControl::slotLoopReset() {
+    setCurrentSavedLoopControl(nullptr);
+}
+
+void CueControl::slotLoopToggled(bool enabled) {
+    if (!m_pCurrentSavedLoopControl) {
+        return;
+    }
+
+    if (enabled) {
```

Comment:
I think it is a good idea to assert if the current loop still match the HoitcueControl before set is Active. 
Maybe we can integrate this into slotLoopUpdated()

Reply:
> I think it is a good idea to assert if the current loop still match the HoitcueControl before set is Active.

Done.

> Maybe we can integrate this into slotLoopUpdated()

Hmm, I think this would make it more complicated. When the loop is re-enabled and didn't changed since the last time it was enabled, the saved loop is set to active again.